### ISSUE TITLE
Ability to use fman as default file viewer on macOS

### ISSUE THREAD

**Issue by matangover** [fman-community/fman#555]
fman is really nice. I want to use it instead of Finder. 😄 
It would be nice to add an option to "set fman as the default file viewer", so that any app that has a "Reveal in Finder" command would open fman instead.

PathFinder has [this functionality](http://support.cocoatech.com/kb/features-and-development/can-i-replace-os-xs-finder-with-path-finder). ForkLift also [support it](https://binarynights.com/manual#fileviewer). I have tried the method used by ForkLift on fman:
```
defaults write -g NSFileViewer -string io.fman.fman
```
However it doesn't work as expected. fman does open (or comes to foreground if already open), but doesn't navigate to the selected file (I tested by clicking "Reveal in Finder" on a downloaded file in Chrome). [This is maybe because fman doesn't handle an "open document" event sent by macOS. This could perhaps be supported by using `QFileOpenEvent` from PyQt.]

Btw this may be possible on Windows/Linux too but that could maybe be a separate feature request.

**Comment by mherrmann** [comment]
@raguay do you have any experience with this?

**Comment by raguay** [comment]
Yes. I've tried everything like is mentioned with the default command. But,
I haven't been able to figure out how to keep it there. The default get
overwritten periodically as well. I haven't been able to figure a solution,
but one must exist since Path Finder does successfully do it. But, I've
noticed that it check on each start as well. So, part of the solution is
for fman to check defaults on startup and set itself if it has been
changed. But, for knowing the passed path, surely it is on the command line?

**Comment by mherrmann** [comment]
I see, interesting that the setting apparently needs to be set repeatedly.

The path should be on the command line and if it is a file, fman does select it.

**Comment by matangover** [comment]
https://github.com/matangover/ReplaceFinder
Works for me™. :) Feedback welcome.
